pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -664,3 +665,21 @@ func enteredCycle(req *internalListUsersRequest) bool {
 	req.visitedUsersetsMap[key] = struct{}{}
 	return false
 }
+
+func (l *listUsersQuery) buildResultsChannel() chan *openfgav1.User {
+	foundUsersCh := make(chan *openfgav1.User, serverconfig.DefaultListUsersMaxResults)
+	maxResults := l.maxResults
+	if maxResults > 0 {
+		foundUsersCh = make(chan *openfgav1.User, maxResults)
+	}
+	return foundUsersCh
+}
+
+func (l *listUsersQuery) trySendResult(ctx context.Context, user *openfgav1.User, foundUsersCh chan<- *openfgav1.User) {
```

Comment:
The `listUsersQuery` isn't really aiding this function. Instead, can we introduce `type foundUserChannel chan<- *openfgav1.User` and use that as the receiver type in the method? 

```suggestion
func (l *foundUsersChan) trySendResult(ctx context.Context, user *openfgav1.User) {
```

Reply:
Adding a new type feels unnecessary. However, I refactored the function.